### PR TITLE
Choking Changes

### PR DESCRIPTION
**Pull request by RSG250**
[Balance]

## About the PR 
Adds a flag to clothing.dm for blocking choking as well as changing neck grab to check for the flag
adds flag to things that would normally block choking as well as most sec gear, yell at me if i forget anything
additionally should fix the issue were you cant choke someone smoking a cigerette

## Why's this needed? 
not being able to choke someone wearing a paper hat is silly.

## Changelog

```
(u)RSG250:
(*)Changed how choking someone is handled, should be able to choke people if it LOOKS like they can be choked
```

### CODE REVIEW

File: _std/defines/clothing.dm [hunk 1]
```diff
@@ -60,6 +60,8 @@
 #define BLOCK_BURN					32768
 /// block an extra point of blunt damage when used to block
 #define BLOCK_BLUNT					65536
+/// blocks choking, also a very silly flag name
+#define UNCHOKE					131072
```
Comment (Tarmunora) [inline _std/defines/clothing.dm:64]: perhaps rename to BLOCKCHOKE, and move it up beside BLOCKSMOKE?

File: code/obj/item/clothing/helmets.dm [hunk 2]
```diff
@@ -492,7 +492,7 @@
 	name = "welding helmet"
 	desc = "A head-mounted face cover designed to protect the wearer completely from space-arc eye. Can be flipped up for clearer vision."
 	icon_state = "welding"
-	c_flags = SPACEWEAR | COVERSEYES
+	c_flags = SPACEWEAR | COVERSEYES | UNCHOKE
```
Comment (Tarmunora) [inline code/obj/item/clothing/helmets.dm:495]: should probably only block choking when flipped down

Comment (RSG250) [reply code/obj/item/clothing/helmets.dm:495]: ah i missed that

File: code/obj/item/clothing/helmets.dm [hunk 3]
```diff
@@ -510,10 +510,12 @@
 		setProperty("disorient_resist_eye", 100)
 
 	proc/flip_down()
+		c_flags = SPACEWEAR | COVERSEYES | UNCHOKE
```
Comment (Tarmunora) [inline code/obj/item/clothing/helmets.dm:513]:
```suggestion
		c_flags |= (COVERSEYES | UNCHOKE)
```

File: code/obj/item/clothing/helmets.dm [hunk 4]
```diff
@@ -510,10 +510,12 @@
...
 		setProperty("meleeprot_head", 2)
 		setProperty("disorient_resist_eye", 100)
 
 	proc/flip_up()
+		c_flags = SPACEWEAR
```
Comment (Tarmunora) [inline code/obj/item/clothing/helmets.dm:518]:
```suggestion
		c_flags &= ~(COVERSEYES | UNCHOKE)
```

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Review by Tarmunora — APPROVED** [review]

File: code/obj/item/clothing/helmets.dm [hunk 5]
```diff
@@ -506,16 +506,16 @@
...
 	proc/flip_down()
-		c_flags |= (COVERSEYES | UNCHOKE)
-		setProperty("meleeprot_head", 2)
+		src.c_flags |= (COVERSEYES | BLOCKCHOKE)
+		setProperty("meleeprot_head", 1)
 		setProperty("disorient_resist_eye", 100)
 
 	proc/flip_up()
-		c_flags &= ~(COVERSEYES | UNCHOKE)
+		src.c_flags &= SPACEWEAR
```
Comment (Tarmunora) [inline code/obj/item/clothing/helmets.dm:518]:
```suggestion
		src.c_flags &= ~(COVERSEYES | BLOCKCHOKE)
```

Comment (RSG250) [reply code/obj/item/clothing/helmets.dm:518]: woundnt that block choking when its flipped up? referencing http://byondref.qtqt.cf/#/operator/&=
it says something like 
A &= B
If A and B are lists, items in A that are not in B are removed. 
witch is the purpose of flip up here?

Comment (RSG250) [reply code/obj/item/clothing/helmets.dm:518]: im probably wrong, but my gut is telling me im right from how testing various ways to do this went

Comment (RSG250) [reply code/obj/item/clothing/helmets.dm]: I was wrong lol

**Review by pali6 — APPROVED** [review]

**Pull request merged**